taurus/entity/object/tests/test_ingredient_spec.py

```diff
@@ -22,3 +29,56 @@ def test_ingredient_reassignment():
     assert potatoes.process == frying
     assert boiling.ingredients == []
     assert set(frying.ingredients) == {oil, potatoes}
+
+
+VALID_QUANTITIES = [
+    NominalReal(14.0, ''),
+    UniformReal(0.5, 0.6, 'm'),
+    NormalReal(-0.3, 0.6, "kg")
+]
+
+INVALID_QUANTITIES = [
+    NominalCategorical("blue"),
+    NominalInteger(5),
+    0.33,
+    "0.5"
+]
+
+
+@pytest.mark.parametrize("valid_quantity", VALID_QUANTITIES)
+def test_valid_quantities(valid_quantity):
+    """
+    Check that all quantities must be continuous values.
+
+    There are no restrictions on the value or the units. Although a volume fraction of -5 kg
+    does not make physical sense, it will not throw an error.
+    """
+    ingred = IngredientSpec(name="name", mass_fraction=valid_quantity)
+    assert ingred.mass_fraction == valid_quantity
```

Comment:
Shouldn't any mass fraction with a dimensional unit be an error?  Or does that qualify as a ValueError instead of a TypeError? (BE responsibility)

Reply:
We've removed that validation from Taurus and put it in the backend. And since Taurus accepts any ContinuousValue, even those that are physically nonsense, I decided to make the tests reflect that (this is explained in the test's docstring)